qtodotxt/test/test_tasks.py

```diff
@@ -182,3 +182,46 @@ def test_creation_date(self):
         self.assertEqual(task.creation_date, date(2016, 3, 15))
         task = Task('x 2017-08-12 do something')
         self.assertFalse(task.creation_date)
+
+    # Recurring tasks
+    # Negative tests
+    def test_recurring_task_wrong_interval(self):
+        task = Task('(B) do something rec:2g')
+        self.assertIsNone(task.recursion)
+
+    def test_recurring_task_wrong_increment(self):
+        task = Task('(B) do something rec:0d')
+        self.assertIsNone(task.recursion)
+
+    def test_recurring_task_wrong_keyword(self):
+        task = Task('(B) do something rect:5d')
+        self.assertIsNone(task.recursion)
+
+    # Positive tests
+    def test_recurring_task_input_days(self):
+        task = Task('(C) do something due:2016-09-05 rec:5d')
+        self.assertIsNotNone(task.recursion)
+        self.assertTrue(task.recursion.mode == recursiveModes.completitionDate)
+        self.assertTrue(task.recursion.increment == str(5))
+        self.assertTrue(task.recursion.interval == 'd')
+
+    def test_recurring_task_input_weeks(self):
+        task = Task('(C) do something due:2016-09-05 rec:+7w')
+        self.assertIsNotNone(task.recursion)
+        self.assertTrue(task.recursion.mode == recursiveModes.originalDueDate)
+        self.assertTrue(task.recursion.increment == str(7))
+        self.assertTrue(task.recursion.interval == 'w')
+
+    def test_recurring_task_input_months(self):
+        task = Task('(C) do something due:2016-09-05 rec:3m')
+        self.assertIsNotNone(task.recursion)
+        self.assertTrue(task.recursion.mode == recursiveModes.completitionDate)
+        self.assertTrue(task.recursion.increment == str(3))
+        self.assertTrue(task.recursion.interval == 'm')
+
+    def test_recurring_task_input_years(self):
+        task = Task('(C) do something due:2016-09-05 rec:+1y')
+        self.assertIsNotNone(task.recursion)
+        self.assertTrue(task.recursion.mode == recursiveModes.originalDueDate)
```

Comment:
I would like to create a task which is recurrent and then complete it and test if another one with the correct due date is there. I cannot get this done. Do you have an idea?